### PR TITLE
Add controller

### LINKED ISSUE #5: newUI: consent form looks horrible
consent-form looks like 💩 on a cracker and needs work

### LINKED ISSUE #10: SSP 2.0: Several translation issues
```
ERROR: UndefinedMethod - www/getconsent.php:212:43 - Method SimpleSAML\Locale\Translate::t does not exist (see https://psalm.dev/022)

    $summary = 'summary="' . $translator->t('List the information about you that is about to be transmitted to the service you are going to login to') . '"';
```

```
ERROR: UndefinedMethod - www/getconsent.php:220:51 - Method SimpleSAML\Locale\Translate::t does not exist (see https://psalm.dev/022)

        $str .= "\n" . '<caption>' . $translator->t('User information') . '</caption>';
```

```
ERROR: UndefinedMethod - www/getconsent.php:277:38 - Method SimpleSAML\Locale\Translate::t does not exist (see https://psalm.dev/022)

                $str .= $translator->t('Show attributes');
```

### PR DESCRIPTION
**Pull request by tvdijen**
Closes #5
Closes #10

### CODE REVIEW

**Comment by tvdijen** [comment]
Needs a workaround for:
https://github.com/simplesamlphp/simplesamlphp-module-consent/blob/controller/lib/Controller/ConsentController.php#L94

Headers already sent

File: lib/Controller/ConsentController.php [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+        // Make, populate and layout consent form
+        $t = new Template($this->config, 'consent:consentform.twig');
+        $translator = $t->getTranslator();
+        $t->data['srcMetadata'] = $state['Source'];
+        $t->data['dstMetadata'] = $state['Destination'];
+        $t->data['yesTarget'] = Module::getModuleURL('consent/getconsent');
+        $t->data['yesData'] = ['StateId' => $stateId];
+        $t->data['noTarget'] = Module::getModuleURL('consent/noconsent');
+        $t->data['noData'] = ['StateId' => $stateId];
```
Comment (thijskh) [inline lib/Controller/ConsentController.php:196]: A lot of these are not used in the template I think... duplications like yesdata/nodata etc

**Review by thijskh — APPROVED** [review]

File: lib/Controller/ConsentController.php [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    /**
+     * Display consent form.
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request The current request.
+     *
+     * @return \SimpleSAML\XHTML\Template|\SimpleSAML\HTTP\RunnableResponse
+     */
+    public function getconsent(Request $request)
+    {
+//        session_cache_limiter('nocache');
```
[the remark this answers is not in the document]
Comment (tvdijen) [reply lib/Controller/ConsentController.php:94]: See my earlier comment..  It's a way to signal the browser that the consent-page may not be cached, but it's not playing nicely with Symfony controllers... However, I've found out that the no-cache flag is the default in Symfony, so we should be fine if we remove this.

**Comment by tvdijen** [comment]
I've found one more issue while fixing your comments... It turns out we had functionality to 'hide' attributes from the consent-screen, meaning they would be collapsed, but could still be viewed when clicked on..

Now that we use the embedded `_table.twig` in favour of the horrible `present_attributes`-method, we've lost this ability.. I'm not sure how to fix this, but I think the easy way out is to wrap another `_table.twig` in an expander and show it at the bottom of the consent-screen.. What do you think about that?

**Comment by thijskh** [comment]
Seems like a fine solution to me.

**Comment by tvdijen** [comment]
I'll see if I can make it happen!

**Comment by tvdijen** [comment]
The fix was actually easier than I thought, because the old behaviour was merely to hide the values.. I simply set them to `null` right before displaying, where the old behaviour was to mark the table cell `display: none`

**Pull request merged**